vm/src/obj/objbyteinner.rs

```diff
@@ -384,14 +551,290 @@ impl PyByteInner {
+        let mut res = vec![];
+
+        let from = match try_as_bytes_like(&from) {
+            Some(value) => value,
+            None => {
+                return Err(
+                    vm.new_type_error(format!("a bytes-like object is required, not {}", from))
+                );
+            }
+        };
+
+        let to = match try_as_bytes_like(&to) {
+            Some(value) => value,
+            None => {
+                return Err(
+                    vm.new_type_error(format!("a bytes-like object is required, not {}", to))
+                );
+            }
+        };
+
+        for i in 0..=255 {
+            res.push(if let Some(position) = from.iter().position(|&x| x == i) {
+                to[position]
+            } else {
+                i
+            });
+        }
+
+        Ok(vm.ctx.new_bytes(res))
+    }
+
+    pub fn translate(&self, options: ByteInnerTranslateOptions, vm: &VirtualMachine) -> PyResult {
+        let (table, delete) = options.get_value(vm)?;
+
+        let mut res = vec![];
+
+        for i in self.elements.iter() {
+            if !delete.contains(&i) {
+                res.push(table[*i as usize]);
+            }
+        }
+
+        Ok(vm.ctx.new_bytes(res))
+    }
+
+    pub fn strip(
+        &self,
+        chars: OptionalArg<PyObjectRef>,
+        position: ByteInnerPosition,
+        vm: &VirtualMachine,
+    ) -> PyResult<Vec<u8>> {
+        let chars = if let OptionalArg::Present(content) = chars {
+            match try_as_bytes_like(&content) {
+                Some(value) => value,
+                None => {
+                    return Err(vm.new_type_error(format!(
+                        "a bytes-like object is required, not {}",
+                        content
+                    )));
+                }
+            }
+        } else {
+            vec![b' ']
+        };
+
+        let mut start = 0;
+        let mut end = self.len();
+
+        if let ByteInnerPosition::Left | ByteInnerPosition::All = position {
+            for (n, i) in self.elements.iter().enumerate() {
+                if !chars.contains(i) {
+                    start = n;
+                    break;
+                }
+            }
+        }
+
+        if let ByteInnerPosition::Right | ByteInnerPosition::All = position {
+            for (n, i) in self.elements.iter().rev().enumerate() {
+                if !chars.contains(i) {
+                    end = self.len() - n;
+                    break;
+                }
+            }
+        }
+        Ok(self.elements[start..end].to_vec())
     }
 }
 
-pub fn is_byte(obj: &PyObjectRef) -> Option<Vec<u8>> {
+pub fn try_as_byte(obj: &PyObjectRef) -> Option<Vec<u8>> {
     match_class!(obj.clone(),
 
     i @ PyBytes => Some(i.get_value().to_vec()),
     j @ PyByteArray => Some(get_value_bytearray(&j.as_object()).to_vec()),
     _ => None)
 }
+
+pub fn try_as_bytes_like(obj: &PyObjectRef) -> Option<Vec<u8>> {
```

Comment:
thanks for the review. This was a problem from beginning for me.
REturning PyResult is the easier. I didn't do it it since I wanted to have strictly the same error messages than cpython (which are very inconsistent) but since it's not strictly   needed PyResult is better.
I answer to your following comment  about many times it could be byutes like or int or None. I hesitate to add a `int' and a '`None' flag to `try_as_bytes_like to directly handle those cases.

About your proposal  of `BytesLike`. I'm not sure to understand : What would be inside that struct ? what would  TryFromObject return ? Vec<u8> ? ByteLikes ? PyByteInner ?

Reply:
> About your proposal of BytesLike. I'm not sure to understand : What would be inside that struct ? what would TryFromObject return ? Vec ? ByteLikes ? PyByteInner ?

I think `PyByteInner` would be great options here.

> I answer to your following comment about many times it could be byutes like or int or None. I hesitate to add a int' and a 'None' flag to `try_as_bytes_like to directly handle those cases.

I agree with what you say. Maybe using `Either<BytesLike, PyIntRef>`?

Reply:
do you mean something like this :
```rust

impl TryFromObject for PyByteInner {
    fn try_from_object(vm: &VirtualMachine, obj: PyObjectRef) -> PyResult<Self> {
        match_class!(obj.clone(),

        i @ PyBytes => Ok(PyByteInner{elements: i.get_value().to_vec()}),
        // j @ PyByteArray => Some(get_value_bytearray(&j.as_object()).to_vec()),
        // k @ PyMemoryView => Some(k.get_obj_value().unwrap()),
        _ => Err(vm.new_type_error(format!(
                            "a bytes-like object is required, not {}",
                            obj
                        ))))
    }
}
```

where would you use this `Either<BytesLike, PyIntRef>` ?

Reply:
What you wrote looks great.

> where would you use this Either<BytesLike, PyIntRef> ?

For example in:

- `ByteInnerFindOptions.sub`
- `contains.needle`

My point here is to move as much as possible of the type check to the shared code already exist.

Reply:
done